tests/test_experimental.py

```diff
@@ -1,14 +1,392 @@
+        smoother = ESMDA(
+            covariance=covariance,
+            observations=observations,
+            alpha=alpha,
+            seed=1,
+        )
+
+        X_i2 = np.copy(X)
+        for i in range(smoother.num_assimilations()):
+            X_i2 = smoother.assimilate(X_i2, g(X_i2))
+
+        assert np.allclose(X_i, X_i2)
+
+    @pytest.mark.parametrize(
+        "linear_problem",
+        range(25),
+        indirect=True,
+        ids=[f"seed-{i+1}" for i in range(25)],
+    )
+    @pytest.mark.parametrize(
+        "cutoffs", [(0, 1e-3), (0.1, 0.2), (0.5, 0.5 + 1e-12), (0.9, 1), (1 - 1e-3, 1)]
+    )
+    def test_that_posterior_generalized_variance_increases_in_cutoff(
+        self, linear_problem, cutoffs
+    ):
+        """As the number of ensemble members decrease, this test starts to fail
+        more often. The property only holds in the limit as the number of
+        ensemble members goes to infinity."""
+
+        # Create a problem with g(x) = A @ x
+        X, g, observations, covariance, rng = linear_problem
+
+        # =============================================================================
+        # SETUP ESMDA FOR LOCALIZATION AND SOLVE PROBLEM
+        # =============================================================================
+        alpha = normalize_alpha(np.ones(1))
+
+        smoother = AdaptiveESMDA(
+            covariance=covariance, observations=observations, seed=1
+        )
+
+        X_i = np.copy(X)
+        for i, alpha_i in enumerate(alpha, 1):
+
+            # Run forward model
+            Y_i = g(X_i)
+
+            # Create noise D - common to this ESMDA update
+            D_i = smoother.perturb_observations(
+                ensemble_size=Y_i.shape[1], alpha=alpha_i
+            )
+
+            cutoff_low, cutoff_high = cutoffs
+            assert cutoff_low <= cutoff_high
+
+            # Update twice, one with low cutoff, once with low cutoff
+            X_i_low_cutoff = smoother.assimilate(
+                X=X_i,
+                Y=Y_i,
+                D=D_i,
+                alpha=alpha_i,
+                correlation_threshold=lambda ensemble_size: cutoff_low,
+            )
+            X_i_high_cutoff = smoother.assimilate(
+                X=X_i,
+                Y=Y_i,
+                D=D_i,
+                alpha=alpha_i,
+                correlation_threshold=lambda ensemble_size: cutoff_high,
+            )
+
+            # Compute covariances
+            prior_cov = np.cov(X, rowvar=False)
+            posterior_cutoff_low_cov = np.cov(X_i_low_cutoff, rowvar=False)
+            posterior_cutoff_high_cov = np.cov(X_i_high_cutoff, rowvar=False)
+
+            # Compute determinants of covariance matrices
+            # https://en.wikipedia.org/wiki/Generalized_variance
+            # intuitively: large determintant => high covariancce
+            #  => smaller volume of multivariate normal
```

Comment:
My intuition is that a large determinant means more spread and thus a larger volume. Am I wrong?